core/src/test/scala/unit/kafka/KafkaConfigTest.scala

```diff
@@ -41,6 +41,23 @@ class KafkaConfigTest {
   @AfterEach
   def tearDown(): Unit = Exit.resetExitProcedure()
 
+  @Test
+  def testRequiredProperties(): Unit = {
+    // process.roles
+    val properties = new Properties()
+    assertBadConfigContainingMessage(properties,
+      "Missing required configuration \"process.roles\" which has no default value.")
+
+    // node.id
```

Comment:
I think this comment is not too helpful - either we must provide more context or just delete it (the error message below makes it clear which config is missing).